### PR TITLE
[C3] Add `getBindingsProxy` support to qwik template

### PR DESCRIPTION
**Pull request by jculvey**
**What this PR solves / how to test:**

Adds the newly introduced `getBindingsProxy` to qwik template. This will allow users to use the built-in `vite dev` that ships with qwik. This was previously blocked by having a reasonable solution to bindings in dev. Similar changes for other frameworks will be coming soon.

As part of this change, we'll be removing `pages:dev` and `pages:deploy` commands in favor of the framework provided `dev` command and a `deploy` command which c3 will generate.

This PR also updates the e2e infrastructure to allow for framework tests to exercise the dev and build commands and verify bindings are working. For now we're using an environment variable binding to avoid having to deal with provisioning/tearing down resources on each run.

It also introduces the notion of fixtures in framework e2e tests. An flat files specified in `e2e-tests/fixtures/framework` will be copied into the test directory after running c3. This allows us to add any additional route handlers or config files we might need to test bindings.

**Author has addressed the following:**

- Tests
  - [ x ] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ x ] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s):
  - [ ] Not necessary because: 

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

### CODE REVIEW

File: packages/create-cloudflare/e2e-tests/frameworks.test.ts [hunk 1]
```diff
@@ -1,249 +1,243 @@
...
-type FrameworkTestConfig = Omit<RunnerConfig, "ctx"> & {
-	expectResponseToContain: string;
+type FrameworkTestConfig = RunnerConfig & {
 	testCommitMessage: boolean;
-	timeout?: number;
 	unsupportedPms?: string[];
 	unsupportedOSs?: string[];
+	verifyDev?: {
+		route: string;
+		expectedToken: string;
```
Comment (dario-piotrowicz) [inline packages/create-cloudflare/e2e-tests/frameworks.test.ts:44]: Nit (and a very personal opinion 😅): I feel like `expectedToken` is a bit misleading and that something like `expectedText` or something like that could be more clear

File: packages/create-cloudflare/templates/qwik/c3.ts [hunk 2]
```diff
@@ -12,24 +16,76 @@ const generate = async (ctx: C3Context) => {
...
 	platform: "pages",
+	devScript: "dev",
+	deployScript: "deploy",
 	generate,
 	configure,
 	transformPackageJson: async () => ({
 		scripts: {
-			"pages:dev": `wrangler pages dev ${await compatDateFlag()} -- ${npm} run dev`,
-			"pages:deploy": `${npm} run build && wrangler pages deploy ./dist`,
+			deploy: `${npm} run build && wrangler pages deploy ./dist`,
```
[the remark this answers is not in the document]
Comment (jculvey) [reply packages/create-cloudflare/templates/qwik/c3.ts:88]: Yes, but at least for `qwik` I'll do it in a follow-up PR. I tried giving it a shot but I'm having some issues getting the existing preview command to work with `wrangler`. `qwik build preview && wrangler pages dev ./dist` doesn't seem to work and I want to understand their preview build process a little more and debug why this isn't working.

File: packages/create-cloudflare/templates/qwik/c3.ts [hunk 3]
```diff
@@ -12,24 +16,76 @@ const generate = async (ctx: C3Context) => {
...
+	const s = spinner();
+	s.start("Updating `vite.config.ts`");
+
+	// Insert the env declaration after the last import (but before the rest of the body)
+	const envDeclaration = `
+let env = {};
+
+if(process.env.NODE_ENV === 'development') {
+  const { getBindingsProxy } = await import('wrangler');
+  const { bindings } = await getBindingsProxy();
```
[the remark this answers is not in the document]
Comment (jculvey) [reply packages/create-cloudflare/templates/qwik/c3.ts:44]: We'll circle back and add these in a separate PR

File: packages/create-cloudflare/e2e-tests/frameworks.test.ts [hunk 4]
```diff
@@ -295,8 +335,158 @@ describe.concurrent(`E2E: Web frameworks`, () => {
...
+
+	// Kill the process gracefully so ports can be cleaned up
+	devProc.kill("SIGINT");
+
+	// Wait for a second to allow process to exit cleanly. Otherwise, the port might
+	// end up camped and cause future runs to fail
+	await sleep(1000);
+
+	// Verify expectation after killing the process so that it exits cleanly in case of failure
+	expect(body).toContain(expectedToken);
```
[the remark this answers is not in the document]
Comment (jculvey) [reply packages/create-cloudflare/e2e-tests/frameworks.test.ts:491]: I do eventually want to refactor this into some shared logic, but I want to add a few more frameworks to see if/how this changes before doing so.

**Review by dario-piotrowicz — APPROVED** [review]
Looks good to me 🎉 

Thanks a lot for the changes! 😃 

___

Just one last small question/suggestion, which we also just be addressed in a followup PR, could/should we include a `wrangler.toml` file similar to the one present in the hello worker with the bindings commented out ready to be uncommented?

**Pull request merged**